pkg/storage/stores/shipper/downloads/table.go

```diff
@@ -232,50 +232,46 @@ func (t *Table) Sync(ctx context.Context) error {
 	return nil
 }
 
-func (t *Table) getOrCreateIndexSet(id string, async bool) (IndexSet, error) {
+// getOrCreateIndexSet gets or creates the index set for the userID.
+// If it does not exist, it creates a new one and initializes it in a goroutine.
+// Caller can use IndexSet.AwaitReady() to wait until the IndexSet gets ready, if required.
+func (t *Table) getOrCreateIndexSet(id string) (IndexSet, error) {
 	t.indexSetsMtx.RLock()
 	indexSet, ok := t.indexSets[id]
 	t.indexSetsMtx.RUnlock()
 	if ok {
 		return indexSet, nil
 	}
+	time.Sleep(time.Second)
```

Comment:
Guess this was left unintentionally from testing?

Reply:
:facepalm: yes, will remove it.